docs/arcaflow/workflows/flow-control.md

```diff
@@ -107,3 +107,179 @@ outputs:
     my_plugin: !expr $.steps.my_plugin.outputs.success
     my_other_plugin: !expr $.steps.my_other_plugin.outputs.success
 ```
+
+### Dry Out with bindConstants()
+
+The builtin function [bindConstants()](expressions.md#functions) allows you to factor out any constant input variables to a foreach subworkflow.
+
+```yaml title="input-repeat-name.yaml"
+iterations:
+  - loop_id: 1
+    name: Matt
+  - loop_id: 2
+    name: Matt
+  - loop_id: 3
+    name: Matt
+  - loop_id: 4
+    name: Matt
+```
+
+```yaml title="workflow-repeat-name.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        iterations:
+          type:
+            type_id: list
+            items:
+              type_id: object
+              id: SubRootObject
+              properties:
+                loop_id:
+                  type:
+                    type_id: integer
+                name:
+                  type:
+                    type_id: string
+
+steps:
+  foreach_loop:
+    kind: foreach
+    items: !expr $.input.iterations
+    workflow: sub-workflow.yaml
+    parallelism: 1
+
+outputs:
+  success:
+    fab_four: !expr $.steps.foreach_loop.outputs.success.data
+```
+
+```yaml title="subworkflow-repeat-name.yaml"
+version: v0.2.0
+input:
+  root: SubRootObject
+  objects:
+    SubRootObject:
+      id: SubRootObject
+      properties: 
+        name:
+          type:
+            type_id: string
+steps:
+  example:
+    plugin:
+      src: quay.io/arcalot/arcaflow-plugin-template-python:all_d5dd9d6
+      deployment_type: image
+    input:
+      name: !expr $.input.name
+
+outputs:
+  success:
+    name: !expr $.steps.example.outputs.success
+```
+
+The input variable `name`'s value is repeated across for each iteration in this input.
```

Comment:
Again, this lead-in doesn't clearly describe what we're about to see. The description *follows it* at line 196. I'd suggest moving that paragraph up before this sentence. We're about to show you how to simplify the schema, and then, here's the new input file, followed by the new workflow.

Reply:
did it